File: javascript/5.js
function range(begin, end) {
  for (let i = begin; i < end; i++) yield i;
}

function gcd(a, b) {
  if (b === 0)
    return a;
  return gcd(b, a % b);
}

function lcm(a, b) {
  return a * (b / gcd(a, b));
}

let result = [i for each (i in range(1, 20))].reduceRight(lcm);
print(result);

File: clojure/5.clj
(defn gcd
  ([a b] (if (zero? b) a (gcd b (rem a b))))
  ([a b & more] (reduce gcd (conj more (gcd a b)))))

(defn lcm
  ([a b] (* a (/ b (gcd a b))))
  ([a b & more] (reduce lcm (conj more (lcm a b)))))

(println (reduce lcm (range 1 20)))
